apis/python/tests/test_experiment_query.py

```diff
@@ -704,41 +694,41 @@ def test_experiment_query_column_names(soma_experiment):
 
 
 def make_dataframe(path: str, sz: int) -> soma.DataFrame:
-    df = soma.DataFrame(uri=path)
-    df.create_legacy(
+    with soma.DataFrame.create(
+        path,
         schema=pa.schema(
             [
                 ("soma_joinid", pa.int64()),
                 ("label", pa.large_string()),
             ]
         ),
         index_column_names=["soma_joinid"],
-    )
-    df.write(
-        pa.Table.from_pydict(
-            {
-                "soma_joinid": [i for i in range(sz)],
-                "label": [str(i) for i in range(sz)],
-            }
+    ) as df:
+        df.write(
+            pa.Table.from_pydict(
+                {
+                    "soma_joinid": [i for i in range(sz)],
+                    "label": [str(i) for i in range(sz)],
+                }
+            )
         )
-    )
-    return df
+    return factory.open(path)
```

Comment:
this is making me think we need a `tiledbsoma.open(uri)` method, published at the top-level...

Reply:
Exposing this as `tiledbsoma.open` is my exact goal once this all goes through.